zephyr/esp32/CMakeLists.txt

```diff
@@ -174,6 +174,13 @@ if(CONFIG_SOC_SERIES_ESP32)
     ../../components/soc/lldesc.c
     )
 
+  zephyr_sources_ifdef(
```

Comment:
I don't think this is a good way to go. You are adding the same sources again for the "different" peripheral.